scipy/ndimage/_backend.py

```diff
@@ -0,0 +1,269 @@
+import scipy._lib.uarray as ua
+from scipy.ndimage import _api
+import numpy as np
+from typing import Type
+
+
+__all__ = ['register_backend', 'set_backend',
+           'set_global_backend', 'skip_backend']
+
+
+class ndimage_output:
+    """
+    Special case outputs to handle multiple types in __ua_convert__.
+    Eg: output='f', output='float32', output=np.float32
+        are all acceptable in addition to arrays.
+    """
+    pass
+
+
+class ndimage_index:
+    """
+    Special case index argument in measurement methods to
+    handle both int and array types in __ua_convert__.
+    """
+    pass
+
+
+class _ScipyImageBackend:
+    """The default backend for ndimage calculations
+
+    Notes
+    -----
+    We use the domain ``numpy.scipy`` rather than ``scipy`` because ``uarray``
+    treats the domain as a hierarchy. This means the user can install a single
+    backend for ``numpy`` and have it implement ``numpy.scipy.ndimage`` as
+    well.
+
+    .. versionadded:: 1.8.0
+    """
+    __ua_domain__ = "numpy.scipy.ndimage"
+
+    @staticmethod
+    def __ua_function__(method, args, kwargs):
+        fn = getattr(_api, method.__name__, None)
+
+        if fn is None:
+            return NotImplemented
+        return fn(*args, **kwargs)
+
+    @ua.wrap_single_convertor
+    def __ua_convert__(value, dispatch_type, coerce):
+        if value is None:
+            return None
+
+        if dispatch_type is np.ndarray:
+            if not coerce and not isinstance(value, np.ndarray):
+                return NotImplemented
+
+            return np.asarray(value)
+
+        if dispatch_type is np.dtype:
+            return np.dtype(value)
+
+        if dispatch_type is ndimage_output:
+            if not isinstance(value, (np.ndarray, Type, np.dtype, str)):
```

Comment:
Is there any particular reason to check `typing.Type` instead of the builtin `type`?

Reply:
Yeah, I should have probably just used type, made it complicated for no reason.
Sorry. Fixed now!